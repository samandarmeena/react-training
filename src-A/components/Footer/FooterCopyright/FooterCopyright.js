import React from 'react';
import PropTypes from 'prop-types';
import styles from './FooterCopyright.module.scss';
import Text from '../../Shared/Text';

/**
 * FooterCopyright component
 * renders Footer copyright container
 * @param {String} copyrightText
 * @param {Array} copyRightLinks
 * @return Element
 */
const FooterCopyright = (props) => {
  const { copyrightText, copyRightLinks } = props;
  return (
    <>
      <div className={styles.copyright}>
        <Text text={copyrightText} />
      </div>
      <div className={styles.TandC}>
        {copyRightLinks.map((link, key) => {
          return <Text key={key} text={link} />;
        })}
      </div>
    </>
  );
};

// proptype validations for props
FooterCopyright.propTypes = {
  copyrightText: PropTypes.string.isRequired,
  copyRightLinks: PropTypes.array.isRequired
};

export default FooterCopyright;
