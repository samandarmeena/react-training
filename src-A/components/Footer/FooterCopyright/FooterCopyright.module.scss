.TandC {
  display: flex;
  justify-content: center;
  span {
    margin-right: 5px;
  }
}
.copyright {
  margin-bottom: 15px;
}
