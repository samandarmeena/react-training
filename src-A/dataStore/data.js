import { IMAGE_URL } from '../constants/constants';
// Footer links array for FooterLinks component
const footerLinkStack = [
  [
    { headerTitle: 'Customer Service' },
    { linkTitle: 'Help Center' },
    { linkTitle: 'Contact Us' },
    { linkTitle: 'Return an Item' },
    { linkTitle: 'Recall Information' },
    { linkTitle: 'Safety Data Sheets' }
  ],
  [
    { headerTitle: 'Corporate Info' },
    { linkTitle: 'Staples Business Advantage Overview' },
    { linkTitle: 'Industries Served' },
    { linkTitle: 'Blog' },
    { linkTitle: 'Our Brands' }
  ],
  [
    { headerTitle: 'Staples Corporate Solutions' },
    { linkTitle: 'Office Products' },
    { linkTitle: 'Technology Products' },
    { linkTitle: 'Furniture' },
    { linkTitle: 'Facilities' },
    { linkTitle: 'Promotional Products' }
  ],
  [
    { headerTitle: 'New Customer' },
    { linkTitle: 'Become a Customer' },
    { linkTitle: 'Govt. Customers' },
    { headerTitle: 'Memberships' },
    { linkTitle: 'Join the Staples Team' },
    { linkTitle: 'Careers' }
  ]
];

// accessibility image for FooterConnect component
const accessibilityImg = {
  src: `${IMAGE_URL}essential-accessibility.png`,
  alt: 'Essential Accessibility'
};

// social connect image for FooterConnect component
const socialImgs = [
  { src: `${IMAGE_URL}linkedin.png`, alt: 'LinkedIn' },
  { src: `${IMAGE_URL}twitter.png`, alt: 'Twitter' },
  { src: `${IMAGE_URL}facebook.png`, alt: 'Facebook' },
  { src: `${IMAGE_URL}youtube.png`, alt: 'Youtube' },
  { src: `${IMAGE_URL}insta.png`, alt: 'Instagram' }
];

// copyright text for footer copyright
const copyrightText =
  'Copyright &copy; 2001-2019 Corporate Express Canada, Inc., a Staples company. All rights reserved.';

// copyright links for footer copyright
const copyRightLinks = [
  'Terms & Conditions',
  ' | ',
  'Privacy Policy',
  ' | ',
  'Staples Bussiness Advantage'
];

// export all data
export {
  footerLinkStack,
  accessibilityImg,
  socialImgs,
  copyRightLinks,
  copyrightText
};
