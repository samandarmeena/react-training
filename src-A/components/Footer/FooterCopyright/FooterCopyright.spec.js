import React from 'react';
import { mount } from 'enzyme';
import FooterCopyright from './FooterCopyright';
import { copyrightText, copyRightLinks } from '../../../dataStore/data';

describe('Checking a child', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = mount(
      <FooterCopyright
        copyrightText={copyrightText}
        copyRightLinks={copyRightLinks}
      />
    );
  });

  it('component rendered without an error', () => {
    expect(wrapper).toBeTruthy();
  });

  it('Check the copyright container is loading', () => {
    const copyrightContainer = wrapper.find('.copyright');
    expect(copyrightContainer.length).toBe(1);
  });

  it('Check the text of copyright container', () => {
    const text = wrapper.find('.copyright').text();
    expect(text).toBe(copyrightText);
  });

  it('Text component should be rendered 5 times', () => {
    const textContainer = wrapper.find('.TandC').find('Text');
    expect(textContainer.length).toBe(5);
    expect(textContainer.first().prop('text')).toEqual('Terms & Conditions');
    expect(textContainer.at(4).prop('text')).toEqual(
      'Staples Bussiness Advantage'
    );
  });

  it('Snapshot testing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
