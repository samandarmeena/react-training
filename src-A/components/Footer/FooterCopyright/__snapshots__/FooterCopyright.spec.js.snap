// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking a child Snapshot testing 1`] = `
<FooterCopyright
  copyRightLinks={
    Array [
      "Terms & Conditions",
      " | ",
      "Privacy Policy",
      " | ",
      "Staples Bussiness Advantage",
    ]
  }
  copyrightText="Copyright &copy; 2001-2019 Corporate Express Canada, Inc., a Staples company. All rights reserved."
>
  <div
    className="copyright"
  >
    <Text
      isStrong={false}
      text="Copyright &copy; 2001-2019 Corporate Express Canada, Inc., a Staples company. All rights reserved."
    >
      <span
        className=""
      >
        Copyright &copy; 2001-2019 Corporate Express Canada, Inc., a Staples company. All rights reserved.
      </span>
    </Text>
  </div>
  <div
    className="TandC"
  >
    <Text
      isStrong={false}
      key="0"
      text="Terms & Conditions"
    >
      <span
        className=""
      >
        Terms & Conditions
      </span>
    </Text>
    <Text
      isStrong={false}
      key="1"
      text=" | "
    >
      <span
        className=""
      >
         | 
      </span>
    </Text>
    <Text
      isStrong={false}
      key="2"
      text="Privacy Policy"
    >
      <span
        className=""
      >
        Privacy Policy
      </span>
    </Text>
    <Text
      isStrong={false}
      key="3"
      text=" | "
    >
      <span
        className=""
      >
         | 
      </span>
    </Text>
    <Text
      isStrong={false}
      key="4"
      text="Staples Bussiness Advantage"
    >
      <span
        className=""
      >
        Staples Bussiness Advantage
      </span>
    </Text>
  </div>
</FooterCopyright>
`;
